pkg/karmadactl/unregister/unregister.go

```diff
@@ -0,0 +1,301 @@
+	if err := cmdutil.DeleteClusterObject(j.ControlPlaneClient, j.ClusterName, j.Wait, j.DryRun); err != nil {
+		klog.Errorf("Failed to delete cluster object. cluster name: %s, error: %v", j.ClusterName, err)
+		return err
+	}
+	klog.Infof("Successfully delete cluster object (%s) from control plane.", j.ClusterName)
+
+	// 2. delete resource created by karmada in member cluster
+	clusterResources := []obj{
+		// 2.1 delete rbac resource which should upload to control plane to serve the controller-manager and aggregated-apiserver
+		{Kind: "ClusterRole", Name: names.GenerateRoleName(names.GenerateServiceAccountName(j.ClusterName))},
+		{Kind: "ClusterRoleBinding", Name: names.GenerateRoleName(names.GenerateServiceAccountName(j.ClusterName))},
+		{Kind: "ServiceAccount", Namespace: j.ClusterNamespace, Name: names.GenerateServiceAccountName(j.ClusterName)},
+		{Kind: "Secret", Namespace: j.ClusterNamespace, Name: names.GenerateServiceAccountName(j.ClusterName)},
+		{Kind: "ServiceAccount", Namespace: j.ClusterNamespace, Name: names.GenerateServiceAccountName("impersonator")},
+		{Kind: "Secret", Namespace: j.ClusterNamespace, Name: names.GenerateServiceAccountName("impersonator")},
+		{Kind: "Namespace", Name: j.ClusterNamespace},
+
+		// 2.2 delete karmada-agent deployment, its own rbac resource and secret/karmada-kubeconfig
+		{Kind: "ClusterRole", Name: register.KarmadaAgentName},
+		{Kind: "ClusterRoleBinding", Name: register.KarmadaAgentName},
+		{Kind: "ServiceAccount", Namespace: j.Namespace, Name: register.KarmadaAgentServiceAccountName},
+		{Kind: "Secret", Namespace: j.Namespace, Name: register.KarmadaConfigSecretName},
+		{Kind: "Deployment", Namespace: j.Namespace, Name: register.KarmadaAgentName},
+	}
+
+	var err error
+	for _, resource := range clusterResources {
+		switch resource.Kind {
+		case "ClusterRole":
+			err = util.DeleteClusterRole(j.MemberClusterClient, resource.Name)
+		case "ClusterRoleBinding":
+			err = util.DeleteClusterRoleBinding(j.MemberClusterClient, resource.Name)
+		case "ServiceAccount":
+			err = util.DeleteServiceAccount(j.MemberClusterClient, resource.Namespace, resource.Name)
+		case "Secret":
+			err = util.DeleteSecret(j.MemberClusterClient, resource.Namespace, resource.Name)
+		case "Deployment":
+			err = deleteDeployment(j.MemberClusterClient, resource.Namespace, resource.Name)
+		case "Namespace":
+			err = util.DeleteNamespace(j.MemberClusterClient, resource.Name)
+		}
+
+		if err != nil {
+			klog.Errorf("Failed to delete (%v) in unregistering cluster (%s): %+v.", resource, j.ClusterName, err)
+			return err
+		}
+		klog.Infof("Successfully delete resource (%v) from member cluster (%s).", resource, j.ClusterName)
+	}
+
+	// 3. delete local obsolete files generated by karmadactl
+	localObsoleteFiles := []obj{
+		{Kind: "File", Name: filepath.Join(register.KarmadaDir, register.KarmadaAgentBootstrapKubeConfigFileName)},
```

Comment:
https://github.com/karmada-io/karmada/blob/c6c20a9e22c3bb96ca487b388318712b7dc4d0c3/pkg/karmadactl/register/register.go#L343-L351
Will file `register.KarmadaAgentBootstrapKubeConfigFileName` be deleted after `karmadactl register` completes?

Reply:
It looks like it will